onnxruntime/test/providers/provider_test_utils.cc

```diff
@@ -1168,8 +1175,12 @@ void OpTester::Run(
         cur_provider = "not set";
       }
 
+#ifdef USE_TENSORRT
+      ORT_UNUSED_PARAMETER(has_run);
```

Comment:
add a comment here to explain this case here.
we are allowing tests to be run with *only* TensorRT EP, but TensorRT EP may not support all tests and may be in excluded providers list.